packages/flyout/src/behaviors/HoverBehavior.js

```diff
@@ -0,0 +1,62 @@
+import { Component } from "react";
+import PropTypes from "prop-types";
+
+const HOVER_DELAY = 500;
+
+export default class HoverBehavior extends Component {
+  static propTypes = {
+    /**
+     * Generally the Flyout target and presenter
+     */
+    children: PropTypes.func.isRequired,
+    /**
+     * Function called when the user enters the visual space
+     * occupied by the component
+     */
+    onMouseEnter: PropTypes.func,
+    /**
+     * Function called when the user leaves the visual space
+     * occupied by the component
+     */
+    onMouseLeave: PropTypes.func
+  };
+
+  constructor(props) {
+    super(props);
+
+    // Binding in the constructor because performance
+    this.handleFocus = this.handleFocus.bind(this);
+    this.handleBlur = this.handleBlur.bind(this);
+
+    this.state = {
+      hasHover: false
+    };
+  }
+
+  handleFocus(event) {
+    if (this.props.onMouseEnter) {
+      this.focusTimeout = setTimeout(() => {
+        this.props.onMouseEnter(event);
+      }, HOVER_DELAY);
+    }
+
+    this.setState({ hasHover: true });
+  }
+
+  handleBlur(event) {
+    if (this.props.onMouseLeave) {
+      clearTimeout(this.focusTimeout);
```

Comment:
The timeout should be cleared regardless of whether `this.props.onMouseLeave` exists.

Reply:
The timeout is only set if the event exists.  Fine w clearing overzealously but curious if there's more to your thinking here

Reply:
@arirusso The timeout is created if `this.props.onMouseEnter` exists, but is only cleared if `this.props.onMouseLeave` exists. Neither of the prop are required so one, both, or none could exist. If the component is only provided with an `onMouseEnter` prop, then timeouts will never be cleared.

Reply:
Updated in 37812694